assets/replace/.github/actions/upgrade/rector.php

```diff
@@ -8,31 +8,46 @@
 use DrupalRector\Set\Drupal9SetList;
 use Rector\Config\RectorConfig;
 use Rector\Core\Configuration\Option;
-use Rector\Core\ValueObject\PhpVersion;
+use Rector\Php80\Rector\FunctionLike\UnionTypesRector;
+use Rector\Php81\Rector\Array_\FirstClassCallableRector;
 use Rector\Set\ValueObject\LevelSetList;
+use Rector\Php80\Rector\FunctionLike\MixedTypeRector;
+
 
 return static function (RectorConfig $rectorConfig): void {
-    // Set desired version support
-    $rectorConfig->sets([
-        Drupal9SetList::DRUPAL_9,
-        LevelSetList::UP_TO_PHP_81,
-    ]);
-
-    $parameters = $rectorConfig->parameters();
-
-    $drupalFinder = new DrupalFinder();
-    $drupalFinder->locateRoot(__DIR__);
-    $drupalRoot = $drupalFinder->getDrupalRoot();
-    $rectorConfig->autoloadPaths([
-        $drupalRoot . '/core',
-        $drupalRoot . '/modules',
-        $drupalRoot . '/profiles',
-        $drupalRoot . '/themes'
-    ]);
-
-    $rectorConfig->skip(['*/upgrade_status/tests/modules/*']);
-    $rectorConfig->fileExtensions(['php', 'module', 'theme', 'install', 'profile', 'inc', 'engine']);
-    $rectorConfig->importNames(true, false);
-    $rectorConfig->importShortClasses(false);
-    $parameters->set('drupal_rector_notices_as_comments', true);
-};
\ No newline at end of file
+  // Set desired version support
+  $rectorConfig->sets([
+    Drupal9SetList::DRUPAL_9,
+    LevelSetList::UP_TO_PHP_81,
+  ]);
+
+  $parameters = $rectorConfig->parameters();
+
+  $drupalFinder = new DrupalFinder();
+  $drupalFinder->locateRoot(__DIR__);
+  $drupalRoot = $drupalFinder->getDrupalRoot();
+  $rectorConfig->autoloadPaths([
+    $drupalRoot . '/core',
+    $drupalRoot . '/modules',
+    $drupalRoot . '/profiles',
+    $drupalRoot . '/themes'
+  ]);
+
+  $rectorConfig->fileExtensions(['php', 'module', 'theme', 'install', 'profile', 'inc', 'engine']);
+  $rectorConfig->importNames(true, false);
+  $rectorConfig->importShortClasses(false);
+  $parameters->set('drupal_rector_notices_as_comments', true);
+
+  $parameters->set(
+    Option::SKIP,
+    [
+      // Don't upgrade array callable to first class callable (not serializable)
+      FirstClassCallableRector::class,
+      // These two rules remove @param for whatever reason
+      MixedTypeRector::class,
+      UnionTypesRector::class,
```

Comment:
The `UnionTypesRector` class has been deprecated in https://github.com/rectorphp/rector/commit/5edc8437c37817781058d845417a3a9441c2968c. What version did you use for testing this?

Reply:
The latest `drupal-rector` release still requires `rector` `v1.15.x`. Their `main` branch however already requires `v1.18.x`. So this will break in the next release. We could still release this config for Drupal 10, but will have to fix it again in the near future.

Reply:
Diese Config habe ich hier gebraucht für die Module: https://github.com/iqual-ch/d10-sw-experiments/blob/main/app/rector.php
Rector 0.15.1: https://github.com/iqual-ch/d10-sw-experiments/blob/main/app/composer.json#L68

Bin für Release und updaten mit 0.18 wenn neuer Release von drupal-rector.

Reply:
Locked: https://github.com/iqual-ch/d10-sw-experiments/blob/main/app/composer.lock#L11891